mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -1235,42 +1238,59 @@ def _get_attributes_filtering_clauses(parsed, dialect):
     return clauses
 
 
-def _to_sqlalchemy_filtering_statement(sql_statement, session, dialect):
-    key_type = sql_statement.get("type")
-    key_name = sql_statement.get("key")
-    value = sql_statement.get("value")
-    comparator = sql_statement.get("comparator").upper()
-
-    if SearchUtils.is_metric(key_type, comparator):
-        entity = SqlLatestMetric
-        value = float(value)
-    elif SearchUtils.is_param(key_type, comparator):
-        entity = SqlParam
-    elif SearchUtils.is_tag(key_type, comparator):
-        entity = SqlTag
-    elif SearchUtils.is_string_attribute(
-        key_type, key_name, comparator
-    ) or SearchUtils.is_numeric_attribute(key_type, key_name, comparator):
-        return None
-    else:
-        raise MlflowException(
-            "Invalid search expression type '%s'" % key_type, error_code=INVALID_PARAMETER_VALUE
-        )
+def _get_sqlalchemy_filter_clauses(parsed, session, dialect):
+    """
+    Creates run attribute filters and subqueries that will be inner-joined to SqlRun to act as
+    multi-clause filters and return them as a tuple.
+    """
+    attribute_filters = []
+    non_attribute_filters = []
 
-    key_filter = SearchUtils.get_sql_comparison_func("=", dialect)(entity.key, key_name)
-    val_filter = SearchUtils.get_sql_comparison_func(comparator, dialect)(entity.value, value)
-    return session.query(entity).filter(key_filter, val_filter).subquery()
+    for sql_statement in parsed:
+        key_type = sql_statement.get("type")
+        key_name = sql_statement.get("key")
+        value = sql_statement.get("value")
+        comparator = sql_statement.get("comparator").upper()
 
+        if SearchUtils.is_string_attribute(
+            key_type, key_name, comparator
+        ) or SearchUtils.is_numeric_attribute(key_type, key_name, comparator):
+            if key_name == "run_name":
+                # Treat "attributes.run_name == <value>" as "tags.`mlflow.runName` == <value>".
+                # The name column in the runs table is empty for runs logged in MLflow <= 1.29.0.
+                key_filter = SearchUtils.get_sql_comparison_func("=", dialect)(
+                    SqlTag.key, MLFLOW_RUN_NAME
+                )
+                val_filter = SearchUtils.get_sql_comparison_func(comparator, dialect)(
```

Comment:
Are there any concerns with minimum length of the search value arg? (do we need to prevent short ILIKE searches on `run_name` (i.e., "run_name ILIKE %model%") to limit the returned materialized results from the tracking db?

Reply:
> do we need to prevent short ILIKE searches on run_name (i.e., "run_name ILIKE %model%")

Does preventing short ILIKE limit the returned materialized results?